test/tesla/middleware/retry_test.exs

```diff
@@ -44,6 +70,10 @@ defmodule Tesla.Middleware.RetryTest do
     assert {:error, :econnrefused} = Client.get("/nope")
   end
 
+  test "use custom retry determination function" do
+    assert {:ok, %{status: 500}} = ClientWithShouldRetryFunction.get("/retry_status")
+  end
```

Comment:
I don't get how does this test work, can you explain?

Reply:
Well... I don't get how it works neither now... What I am writing originally... What an idiot XD.  I will update the unit test. Please wait for it. Thank you!